src/components/client/data/Data.tsx

```diff
@@ -61,18 +69,55 @@ export const Data: React.FC<{ data: IInstructionData }> = ({
           });
         }}
       >
-        <TabList>
-          <Tab {...tabStyle} rounded="md">
-            Borsh{" "}
-            {isAnchor && <LockIcon ml="1" w="2.5" color={lockedIconColor} />}
-          </Tab>
-          <Tab {...tabStyle} rounded="md" isDisabled={isAnchor}>
-            Buffer Layout
-          </Tab>
-          <Tab {...tabStyle} rounded="md" isDisabled={isAnchor}>
-            Raw
-          </Tab>
-        </TabList>
+        <Box display="flex">
+          <TabList>
+            <Tab {...tabStyle} rounded="md">
+              Borsh{" "}
+              {isAnchor && <LockIcon ml="1" w="2.5" color={lockedIconColor} />}
+            </Tab>
+            <Tab {...tabStyle} rounded="md" isDisabled={isAnchor}>
+              Buffer Layout
+            </Tab>
+            <Tab {...tabStyle} rounded="md" isDisabled={isAnchor}>
+              Raw
+            </Tab>
+          </TabList>
+          {
+            format === "raw" ? <Box ml="auto">
+              <ToggleIconButton
+                ml="1"
+                size="sm"
+                label={isHex ? "Click to use encoded data" : "Click to use hex data"}
+                icon={<Icon as={HiOutlineSwitchHorizontal} />}
+                isDisabled={isAnchor}
+                toggled={!isHex}
+                onToggle={(toggled) => {
+                  update((state) => {
```

Comment:
Once your inline functions get this long, you can move them to the top, before the `return`.

Reply:
refactored